barretenberg/cpp/src/barretenberg/protogalaxy/protogalaxy_prover.cpp

```diff
@@ -86,7 +86,10 @@ std::shared_ptr<typename ProverInstances::Instance> ProtoGalaxyProver_<ProverIns
             }
         });
     }
-    next_accumulator->prover_polynomials = std::move(acc_prover_polynomials);
+    for (auto [next_acc_poly, acc_poly] :
+         zip_view(next_accumulator->proving_key->get_all(), acc_proving_key_polys.get_all())) {
+        next_acc_poly = std::move(acc_poly);
```

Comment:
why do you have to loop here?

Reply:
Proving key has some other stuff like circuit_size and evaluation_domain and whatever. This is just looping over the polynomials

Reply:
yeah it's kinda ugh that we create an incomplete proving key :-?

Reply:
but i guess it's just an intermediary object because we carry over the  prover polynomials

Reply:
yeah, it's ugly but I think I can handle the refactoring of this function in a future PR by setting the next accumulator to just be instances[0] and then folding in the rest of the instances instead of creating empty stuff and doing the folding.

Reply:
sounds good!! I approved anyways